### DISCUSSION TITLE
How to use it with custom CNAMEs?

### DISCUSSION THREAD

**Discussion by miticm** [cloudinary-community/next-cloudinary#208] [General]
Since I only pass public Id to the cldimage, I get the res.clouldinary.com always, is there a way to use https://www.example.com/image/upload/sample.jpg instead?  Or is this configured from the Cloudinary side?

**Comment by colbyfayock** [comment]
hey @miticm - i just added an update to allow for this

the current way to support this is by using a `config` prop:

```
<CldImage
  ...
  config={{
    url: {
      secureDistribution: 'spacejelly.dev',
    }
  }}
/>
```

it allows for the parameters defined here: https://cloudinary.com/documentation/cloudinary_sdks#configuration_parameters

id like to provide a way for this to be configured in a global scope, but still thinking on the best and most reliable way to approach that

**Reply by miticm** [reply]
Thanks! 
Yeah, a global scope config would be great, since I would want to use it in cldvideoplayer as well

**Reply by miticm** [reply]
Hi @colbyfayock , since global config is not recommended by nextjs team, can we add the config option to cldVideoPlayer as well? Or do we have a new fix for it that I missed? Thanks!

**Reply by colbyfayock** [reply]
@miticm yeah looks like it's missing, ill try to get that in today or tomorrow

**Reply by colbyfayock** [reply]
@miticm now available via: https://github.com/colbyfayock/next-cloudinary/pull/348

https://next.cloudinary.dev/cldvideoplayer/configuration#config--delivery

at some point i want to normalize the options and how they're passed in between the different components but i need to save that for another day, as thats a can of worms. this just exposes the options that were on the player config

for now, you'll want to pass in the following if using `secure: true` (default):
```
secureDistribution="mycname"
privateCdn={true}
```

or if you're explicitly setting `secure: false`:
```
cname="mycname"
privateCdn={true}
```

**Reply by miticm** [reply]
@colbyfayock Thanks for the updates, but the queryParams is missing, I need to use it as well for the video, for CldImage I can use it with         
```
config={{
          url: {
            queryParams: "my-params",
          },
   }}
```

**Reply by colbyfayock** [reply]
hey @miticm merged in the additional type to support queryParam, should be able to pass it in as a top level option

```
queryParams={{
  myParam: 'value'
}}
```

https://next.cloudinary.dev/cldvideoplayer/configuration#config--delivery

i tested this out locally and the object syntax seems tow ork as expect but not the string, but perhaps im not familiar with how it works, as it seems to split the string

however looking at the type definition on URL Gen SDK it should support it, so not sure if thats a bug or if again, i dont undersatnd how the string format works in this instance

example:
```
https://res.cloudinary.com/colbycloud-next-cloudinary/video/upload/q_auto/vc_vp9/v1/videos/mountain-stars.webm?0=m&1=y&2=-&3=p&4=a&5=r&6=a&7=m&8=s&9==&10=t&11=r&12=u&13=e&14=&&15=t&16=e&17=s&18=t&19==&20=a&21=s&22=d&23=f&_s=vp-1.10.1
```
